joplin/importer/page_importer.py

```diff
@@ -20,41 +21,40 @@
 
 class PageImporter:
     def create_page(self):
-        page_creator_dictionary = {
-            'topiccollection': create_topic_collection_page_from_page_dictionary,
-            'topic': create_topic_page_from_page_dictionary,
-            'information': create_information_page_from_page_dictionary,
-            'services': create_service_page_from_page_dictionary,
+        page_creators = {
+            'topiccollection': create_topic_collection_page_from_importer_dictionaries,
+            'topic': create_topic_page_from_importer_dictionaries,
+            'information': create_information_page_from_importer_dictionaries,
+            'services': create_service_page_from_importer_dictionaries,
         }
 
-        page = page_creator_dictionary[self.page_type](self.page_dictionary, self.revision_id)
+        page = page_creators[self.page_type](self.page_dictionaries, self.revision_id)
 
         return page
 
     def fetch_page_data(self):
-        sample_transport = RequestsHTTPTransport(
-            url=self.joplin_api_endpoint,
-            # todo: use headers to get different languages
-            # headers={
-            #     "Content-type": "application/json",
-            # },
-            verify=True
-        )
-
-        client = Client(
-            retries=3,
-            transport=sample_transport,
-            fetch_schema_from_transport=True,
-        )
-
-        result = client.execute(queries[self.page_type], variable_values=json.dumps({'id': self.revision_id}))
-        revision_node = result['allPageRevisions']['edges'][0]['node']
-
-        # this gets us into the 'as____Page' stuff
-        page_dictionary_from_revision = next(iter(revision_node.values()))
-
-        # set the page dictionary on ourselves but also return it
-        self.page_dictionary = page_dictionary_from_revision
+        # todo: don't just hardcode lang here
```

Comment:
why not?

Reply:
I figure this is one of those lingering todos, where we have future hope for a time when we can decide exactly which languages we want to import

for now it stays todo